### ISSUE TITLE
Colors: Figma and code tokens don't match

### ISSUE THREAD

**Issue by cseas** [razorpay/blade#1312]
Designers have decided to use a different naming convention for color tokens in Figma compared to what was approved in the tokens RFC. 

This is causing a lot of confusion for developers because the tokens that designers are using in Figma have no direct equivalent in code, instead the developers have to guess what the right token is by jumbling up the Figma token's words.

*Original issue report which helped identify this problem is below for reference.*

> ## Problem
>
> The icon components don't accept arbitrary values for color.
>
> Below code throws a TypeScript error.
>
> ```tsx
> <CheckIcon color="#32ED93" size="medium" />
> ```
>
> Required for website [Footer design][1]:
>
> <img width="488" alt="Screenshot 2023-06-16 at 2 08 30 AM" src="https://github.com/razorpay/blade/assets/29686866/1bad7552-cf53-45e2-828c-772cc0b99c7f">
>
> ## Suggestion
>
> Either add support for arbitrary values for the color prop or add support for using the Emerald color palette which comes closest to the above design.
>
> [1]: https://www.figma.com/file/ytpVeqLD5hQOb1WwrdjBsU?node-id=695:120741#473272610

**Comment by anuraghazra** [comment]
@saurav12 need design inputs here. 

@cseas ideally this section can have tweaks in the design to use blade icons and appropriate icon colors (eg: a white icon if emerald isn't present)

**Comment by cseas** [comment]
After discussion with designer, we've replaced it with the token: `feedback.positive.action.icon.link.active.highContrast`

Small feedback: It was difficult to cross reference this token from Figma.

On Figma, the token is: `Feedback/Action/Link/Icon/Positive/highContrast/Active`
In code, the token is: `feedback.positive.action.icon.link.active.highContrast`

Why are terms jumbled up between the two? Is this intended?

**Comment by anshulsahni** [comment]
@anuraghazra can you tell the reason for this discrepancy here?

**Comment by anuraghazra** [comment]
This is intentional, Forwarding rastogi's comment on this: 

@saurav12 

> This is because the grouping on figma will become complex if we follow the same structure and this is intentional.

> In the case you shared, if we group things based on the intent, then the grouping will be made on the basis of intent but we didn't want that. As a design consumer we want them to see the major categories first (Surface, Brand, Feedback & Action) and that's why we decided to make this difference.
> Dev didn't need to go through the tree structure and select a color but on design designers need to search through a lot of colors.

**Comment by cseas** [comment]
What's the dev guideline to figure out the token for a Figma color then? We can't play scrabble for every token please. 😢

**Comment by anuraghazra** [comment]
We will discuss this and see if we can resolve this somehow. 

There is a [how to page](https://blade.razorpay.com/?path=/docs/guides-how-to-use--page#mapping-tokens-from-figma-to-blade-in-your-code) on blade, but it doesn't cover this case. 

There is a pattern I can see, but I'll confirm once if this pattern is consistent across all the tokens

<img width="453" alt="image" src="https://github.com/razorpay/blade/assets/35374649/399a5c03-1d37-4f81-b288-5bae688462fb">

**Comment by cseas** [comment]
> There is a [how to page](https://blade.razorpay.com/?path=/docs/guides-how-to-use--page#mapping-tokens-from-figma-to-blade-in-your-code) on blade, but it doesn't cover this case.

The linked docs mention:
> Look at the component properties and copy it in code because the same properties exists on the components shipped.

Looks like this is not being followed in the current Figma setup. Ordering of tokens was already decided in the tokens RFC. No reason to suddenly change it now. And if designers want to change the ordering, devs should get the changed ordering too.

**Comment by anuraghazra** [comment]
@cseas we will discuss this and look for a long term solution but right now we won't be prioritising this since it will be a huge task to change anything on either design or dev side. 

Meanwhile as an interim solution we will document this pattern on the "how to use" guideline page.

**Comment by cseas** [comment]
Can we please provide a table for the token mapping then? Doing a Ctrl+F would be much faster than reasoning about pattern logic.

**Comment by cseas** [comment]
Another example:

```tsx
          // Figma: Feedback/Action/Link/Icon/Info/highContrast/Disabled
          backgroundColor: paymentTheme.colors.onLight.feedback.icon.information.highContrast,
```

- Tried to guess the right token using TypeScript intellisense
- I was hoping to see a bluish color, this turned out to be white

```tsx
          // Figma: Feedback/Action/Link/Icon/Info/highContrast/Disabled
          backgroundColor: paymentTheme.colors.onLight.action.icon.link.disabled,
```

- Tried again
- This isn't the right token either

```tsx
          // Feedback/Action/Link/Icon/Info/highContrast/Disabled
          backgroundColor:
            paymentTheme.colors.onLight.feedback.information.action.border.primary.disabled
              .highContrast,
```

Finally just copied the hsla value of the token from Figma and did a `Ctrl + F` on Storybook. Found 6 results. Finally decided on one and entered the token referencing every word from Storybook.

It's the right color now but I'm still not sure whether it's the right token because the Figma one has the word "Icon" and the one I used doesn't have it anywhere.

Finding tokens used on Figma is becoming very confusing. Please align the designers to use the same tokens as code or provide a map we can search through for reference.

[Figma][1]

[1]: https://www.figma.com/file/ytpVeqLD5hQOb1WwrdjBsU/PP%2C-PB---Blade-DS-adoption?type=design&node-id=1296-142778&mode=design&t=vvq3PAWGz4khigKK-4

**Comment by kamleshchandnani** [comment]
As explained in the workshop the reason why this happens is because if the shortcoming of Figma and this will be addressed while we migrate to `variables` on Figma which is already planned. For now, that’ll be the case for background colors and custom  things but it's just a value swap between object keys

**Comment by cseas** [comment]
Makes sense. Let's keep the issue open till the variables migration is finished. This isn't a major blocker.

> it's just a value swap between object keys

Still not sure about this though. As shown in the above example:
- Figma: Feedback/Action/`Link`/`Icon`/Info/highContrast/Disabled
- Code: feedback.information.action.`border`.`primary`.disabled.highContrast

I've highlighted the words that don't match between the two.

✅ Both tokens map to the same hsla value.
✅ Both tokens have *most* words the same.

There probably is a token that matches all the words but that's the issue I'm trying to highlight. Finding the right token with IntelliSense is *very difficult* because we don't know which key to begin with.

❌ When I took "feedback" as the first key, I ended up with `feedback.icon.information.highContrast`. Wrong token.
❌ When I took "action" as the first key, I ended up with `action.icon.link.disabled`. Wrong token again.

Searching the hsla value in Storybook is the best method devs have available right now that can get them to the right *color* the fastest but the token they end up using is probably incorrect semantically, which might hamper the design in future if Blade updates the tokens at some point.

Would be great if it's possible to have some guideline for the mapping to minimise the accidental use of wrong tokens. If not, we can wait for the variables migration since it's not a burning issue.
